macos/FluentUI/Button/Button.swift

```diff
@@ -253,6 +253,43 @@ open class Button: NSButton {
 		let path = NSBezierPath(roundedRect: bounds, xRadius: cornerRadius, yRadius: cornerRadius)
 		path.fill()
 	}
+	
+	/// Indicates if the Window that the button view has been added to, is inactive/backgrounded
+	private var isWindowInactive: Bool? {
+		didSet {
+			guard oldValue != isWindowInactive else {
+				return
+			}
+			// Re-compute the Button's color values for the latest Window State, and re-render it
+			setColorValues(forStyle: style, accentColor: accentColor, isWindowInactive: isWindowInactive ?? false)
+			needsDisplay = true
+		}
+	}
+
+	public override func viewDidMoveToWindow() {
+		super.viewDidMoveToWindow()
+
+		// Hook in Notification Handles to capture the Window's active and inactive states
+		NotificationCenter.default.addObserver(forName: NSWindow.didResignMainNotification,
+											   object: nil,
```

Comment:
We ned to be listening to the actual window we're moving to. Otherwise we'll react to notifications for all windows